components/org.wso2.carbon.identity.media.endpoint/src/main/java/org/wso2/carbon/identity/media/endpoint/common/MediaEndpointDataHolder.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.media.endpoint.common;
+
+import org.wso2.carbon.user.core.service.RealmService;
+
+/**
+ * OSGi service holder class for media service.
+ */
+public class MediaEndpointDataHolder {
+
+    private static MediaEndpointDataHolder instance = new MediaEndpointDataHolder();
+    private RealmService realmService;
+
+    private MediaEndpointDataHolder() {
+
+    }
+
+    public static MediaEndpointDataHolder getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Get RealmService osgi service.
+     *
+     * @return RealmService
+     */
+    public RealmService getRealmService() {
+
+        return MediaEndpointDataHolder.getInstance().realmService;
```

Comment:
no need to calll MediaEndpointDataHolder.getInstance() here.. we can directly return realmService